src/Func.ls

```diff
@@ -12,8 +12,15 @@ fix = (f) ->
 
 over = (f, g, x, y) --> f (g x), (g y)
 
-#? memoize, wrap
+memoize = (f) ->
+  memo = {}
+  ->
+    args = Array::slice.call &, 0
+    key = args.join ''
```

Comment:
This just feels too primitive to me. It would be easy to force a name clash. I feel that it might be better using some sort of unordered, ES6-style map or similar for the memo, instead of a plain object. Here's the current problem:

```ls
f = memoize (x) -> x
g = memoize (x) -> x

f 'foo,bar'   #=> 'foo,bar'
f <[foo bar]> #=> 'foo,bar'

g <[foo bar]> #=> <[foo bar]>
g 'foo,bar'   #=> <[foo bar]>
```

Fixing this would be nice, but if this one doesn't fix it, I may send one later.